clients/pkg/promtail/scrapeconfig/scrapeconfig.go

```diff
@@ -202,6 +202,9 @@ type SyslogTargetConfig struct {
 	// message should be pushed to Loki
 	UseRFC5424Message bool `yaml:"use_rfc5424_message"`
 
+	// IsRFC3164Message defines wether the log is formated as RFC3164
+	IsRFC3164Message bool `yaml:"is_rfc3164_message"`
+
```

Comment:
Sorry @catap I thought I left this comment yesterday:

Is there a way we can just detect whether each message is RFC3164 or RFC5424 format, rather than adding another config option? Is it possible that a single target could output logs in both formats?

Reply:
@cstyan nope, it is impossible to determine which format is used. Usually sender sends in supported format, or it may have a selector where someone may choose the format.

The best that I can do here is use `handleMessageRFC3164` as fallback. If `RFC5424` can't parse it, let use `RFC3164` but it may lead to the case when `RFC5424` parses it wrong way, so, I suggest to avoid that.

Reply:
Okay that's fine. 

Lets keep the logic as is then, except can we change the config option to be `SyslogFormat` which has multiple string possibilities, like one of `RFC5425, RFC3164, another RFC, etc.`? And then have a validation function when creating the loading the syslog config or creating the syslog target. 

And it's default value can be `RFC5424`. That way we don't need to add another config field if at some point the future we add support for yet another format. it just has to be another possible option for the validation function.

We also need a reference for this in the syslog section of `docs/sources/send-data/promtail/configuration.md`

Reply:
Here the catch: right now we have two options:
```
	// UseRFC5424Message defines whether the full RFC5424 formatted syslog
	// message should be pushed to Loki
	UseRFC5424Message bool `yaml:"use_rfc5424_message"`

	// IsRFC3164Message defines wether the log is formated as RFC3164
	IsRFC3164Message bool `yaml:"is_rfc3164_message"`
```
and I feel that this should be reworked as weel. To something like:
 - SyslogFormat: RFC5424 (default) or RFC3164;
 - UseFullSyslogMessage: default is false

Reply:
> and I feel that this should be reworked as weel. To something like:
> 
> SyslogFormat: RFC5424 (default) or RFC3164;
> UseFullSyslogMessage: default is false

this makes sense to me, we would have to deprecate `UseRFC5424Message` for now and have the actual value for `use syslog` be an or of `UseRFC5424` and `UseFullSyslogFormat: false`  when `SyslogFormat` is `RFC5424`

I can present your PR to the team as a "we should do this as a bug fix, it also technically requires these config changes" and we should be able to get it merged

Reply:
@cstyan I was wrong, RFC3164 hasn't got string method and implement it isn't something that seems easy. So, let keep it as is.